Add migration support

nklomp opened the pull request:
Implement migrations. Also had to update most entities from the data-store to make them compatible with both sqlite and postgres.

Draft, since the mnemonic seed module needs similar work

nklomp commented:
Ready for review:

Please note that current version is ubstable.70, which is lower than an already published .92. Has to do with CI/CD being confused by a branch being replaced by me. Any code depending on this PR will need an update to a -next version after this has been merged to develop

BtencateSphereon commented on packages/data-store-common/LICENSE:
```diff
@@ -0,0 +1,201 @@
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright 2019 Consensys AG
```
nitpicking here, should this not be 2022 and Sphereon BV?

nklomp replied:
Hehe, guess someone copied some stuff over :D

BtencateSphereon commented on packages/data-store-common/src/__tests__/entities.test.ts:
```diff
@@ -0,0 +1,145 @@
+      synchronize: false,
+      entities: DataStoreConnectionEntities,
+    })
+    // await dbConnection.dropDatabase()
```
this can be removed?

nklomp replied:
Yeah will remove the dropDatabase. If you don't mind I will leave the logging comment. This allows someone to enable it quickly in case of problems with the test

nklomp commented:
Addressed the comments. Also ensured we actually apply the migrations to our internal reference DBs. Added a bit of documentation about that as well in the READMEs.

BtencateSphereon reviewed (approved):
LGTM
